src/main/java/org/apache/maven/plugins/deploy/DeployMojo.java

```diff
@@ -315,6 +336,95 @@ else if ( !project.getAttachedArtifacts().isEmpty() )
         return request;
     }
 
+    private class DeployedPomChecker
+    {
+        final Artifact pomArtifact;
+        final DeployRequest request;
+        final RemoteRepository repo;
+
+        DeployedPomChecker( Artifact pomArtifact, DeployRequest request )
+        {
+            this.pomArtifact = pomArtifact;
+            this.request = request;
+            this.repo = request.getRepository();
+        }
+
+        public void addPomConditionally ()
+        throws MojoExecutionException, MojoFailureException
+        {
+            if ( comparePomWithDeployed )
+            {
+                try
+                {
+                    if ( deployedPomIsEqual() )
```

Comment:
Why not just hash the two and compare hashes?

Reply:
Oi! You are fast! Thanks!

I was just following "orders" from the [MDEPLOY-118](https://issues.apache.org/jira/browse/MDEPLOY-118): "modulo platform-dependent line-break concerns, and that's important!". Then, I thought, why not compare them semantically.

But I was not at all sure, if `Xpp3Dom` is the right way to go. That would have been one of the questions I intended to write in the description of this PR...

Reply:
@cstamas You cannot expect the file to be portable. This isn't a binary file. It is test with line endings. For now, I don't see a better way than @mdrie has done it. I wonder wether this would still fail if `build.output.timestamp` has been modified...

Reply:
Well, yes, if different OS-es are involved. No, if not. Hence, am thinking, maybe introduce different "strategies" to compare?
* parse/model wise (as now\
* checksum wise (just checksum the two "blobs" as byte stream) - maybe some normalization here? Or "checksum text" as XML is text/plain, so read in the lines only, checksum that.... 
* maybe even just some property/version wise (ie. CI friendly version contains git hash, if hashes same....is same commit)
etc

Reply:
Hey Tamás,

I am not quite sure, how to proceed with your comment / request. For the use case in the issue, I think, the "model wise" compare seems to be optimal, since its whole purpose is to deploy from several machines/platforms which might have different line end encodings. "checksum wise" probably has a  performance advantage... but does that really matter here?

The "per property" approach is nice, but I would need a better understanding of the use-case and how it differs from my use-case (as described in the issue), and how a user might want to configure it.

Thinking of CI-friendly: We do use the flatten-maven-plugin for that and I expect the flattened POM to be compared. Therefore, `${revision}` and the like will be resolved and a git hash will be included in the flattened POM. However, there might still be other differences in the flattened POM due to differences in the invocation of Maven.

My thinking was, that this feature ensures, that the deployed POM correctly describes all the deployed artifacts attached to it. If you allow differences in the POM with a more allowing mode, this guarantee is broken.

However, I am not that experienced in the Maven world! Please, tell me, where my thinking is wrong.

Reply:
No, i was just tinkering... just go along with your goal as is, that above may be something to be done is some othre PR (improvement) of this feature.

Reply:
Ok, closing this conversation.